src/buildercore/terraform.py

```diff
@@ -564,6 +569,230 @@ def render_bigquery(context, template):
 
     return template.to_dict()
 
+def render_eks(context, template):
+    if not context['eks']:
+        return {}
+
+    _render_eks_master(context, template)
+    _render_eks_workers(context, template)
+
+def _render_eks_master(context, template):
+    # all from https://learn.hashicorp.com/terraform/aws/eks-intro
+
+    template.populate_resource('aws_eks_cluster', 'main', block={
+        'name': context['stackname'],
+        'version': context['eks']['version'],
+        'role_arn': '${aws_iam_role.master.arn}',
+        'vpc_config': {
+            'security_group_ids': ['${aws_security_group.master.id}'],
+            'subnet_ids': [context['eks']['subnet-id'], context['eks']['redundant-subnet-id']],
+        },
+        'depends_on': [
+            "aws_iam_role_policy_attachment.master_kubernetes",
+            "aws_iam_role_policy_attachment.master_ecs",
+        ]
+    })
+
+    template.populate_resource('aws_iam_role', 'master', block={
+        'name': '%s--AmazonEKSMasterRole' % context['stackname'],
+        'assume_role_policy': json.dumps({
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Principal": {
+                        "Service": "eks.amazonaws.com"
+                    },
+                    "Action": "sts:AssumeRole"
+                }
+            ]
+        }),
+    })
+
+    template.populate_resource('aws_iam_role_policy_attachment', 'master_kubernetes', block={
+        'policy_arn': "arn:aws:iam::aws:policy/AmazonEKSClusterPolicy",
+        'role': "${aws_iam_role.master.name}",
+    })
+
+    template.populate_resource('aws_iam_role_policy_attachment', 'master_ecs', block={
+        'policy_arn': "arn:aws:iam::aws:policy/AmazonEKSServicePolicy",
+        'role': "${aws_iam_role.master.name}",
+    })
+
+    template.populate_resource('aws_security_group', 'master', block={
+        'name': 'project-with-eks--%s--master' % context['instance_id'],
+        'description': 'Cluster communication with worker nodes',
+        'vpc_id': context['aws']['vpc-id'],
+        'egress': {
+            'from_port': 0,
+            'to_port': 0,
+            'protocol': '-1',
```

Comment:
> NOTE: Setting protocol = "all" or protocol = -1 with from_port and to_port will result in the EC2 API creating a security group rule with all ports open. This API behavior cannot be controlled by Terraform and may generate warnings in the future.

https://www.terraform.io/docs/providers/aws/r/security_group_rule.html

if the intent is to open all ports up on all interfaces, could we get a comment to that effect, please, and something about why it isn't a security issue (internal traffic I presume? blocked by VPC/subnet rules?)

Reply:
This is `egress`, it is traffic going from the cluster to the outside of it.